### ISSUE TITLE
Port to Firefox

### ISSUE THREAD

**Issue by smaragdus** [stefanXO/Tab-Manager-Plus#11]
Hello there,

First, I would like to thank you for Sync Tab Groups extension- I find it very useful.

I would like to ask- do you have plans to port Tab Manager Plus to Firefox? I have tested some similar extensions  for Firefox but I have discarded all of them since for me none of them comes even close to this one. I mostly use Chromium forks but sometimes I use Firefox forks so I prefer to use cross browser extensions in order to have similar functionality in different browsers. I suppose that many users would be happy if Tab Manager Plus is ported to Firefox.

Regards

**Comment by mastef** [comment]
We can try @smaragdus

**Comment by smaragdus** [comment]
@mastef 

I am not a developer so I will be glad if you find time and motivation to try. ;)
At least (as far as I know) Mozilla won't charge you for an account.

**Comment by Gavrilian** [comment]
Yes Please! I absolutely love the ui this has, and I was very disappointed to find there wasn't a firefox build, so I second this request!

**Comment by mastef** [comment]
ok I'll take an hour today to take a look if that's feasible

**Comment by mastef** [comment]
@smaragdus @Gavrilian you can try to load this as a debug add-on, and then let me know if it works for you correctly : [tab-manager-plus-5.0.0.firefox.zip](https://github.com/stefanXO/Tab-Manager-Plus/files/3466336/tab-manager-plus-5.0.0.firefox.zip)

**Comment by smaragdus** [comment]
@mastef 

Thank you very much. I will install and test the Firefox extension tonight.

**Comment by smaragdus** [comment]
@mastef 

I could not test **Tab Manager Plus** [version](https://github.com/stefanXO/Tab-Manager-Plus/releases) [5.0.0](https://github.com/stefanXO/Tab-Manager-Plus/releases/tag/5.0.0) because I could not install it- when I tried to install the extension I got the following error message:

`This add-on could not be installed because it appears to be corrupt.`

**screen**

![Tab Manager Plus 5 0 0 - 2019-08-06 - error - 001](https://user-images.githubusercontent.com/9025757/62501994-fa8e2180-b7f5-11e9-8473-e83d22e4677e.png)

I am using a custom build of **Firefox**- [Iceweasel](https://sourceforge.net/projects/libportable/files/Iceweasel/Stable/).

**Comment by mastef** [comment]
@smaragdus You need to install it through the "Debug Add-ons" section, and then "Load Temporary Add-On..."

![image](https://user-images.githubusercontent.com/762297/62506883-8b7fef80-b82c-11e9-8c7a-85d33b4095f7.png)

The direct link is `about:debugging#addons`

**Comment by smaragdus** [comment]
@mastef 

Thank you for the detailed tip and the screen, I really did not know that I should add the extension via 'Debug Add-ons'. I will try to test the extension tonight.

**Comment by mastef** [comment]
@smaragdus @Gavrilian `Tab Manager Plus for Firefox` has been approved now by the mozilla team and is available on the official Firefox Add-Ons page : https://addons.mozilla.org/en-US/firefox/addon/tab-manager-plus-for-firefox/

**Comment by smaragdus** [comment]
@mastef 

**Tab Manager Plus** version **5.0.1** does not work with my Chromium-based [Cent Browser](https://www.centbrowser.com/) ([version 4.0.9.112](https://www.centbrowser.com/history.html)), the tool bar icon is there but it is inaccessible- it does not show the number of open tabs, I cannot open Tab Manager Plus window, options window is blank. This is a very serious problem. Shall I open a new issue? **Tab Manager Plus** version **4.9.9** was working fine but version **5.0.1** does not work at all.

**Comment by mastef** [comment]
@smaragdus Yeah, it's kind of related to the Firefox changes. 5.0.2 is out now and should be available in a few minutes for you. That should make it work again - sorry for the inconvenience

**Comment by smaragdus** [comment]
@mastef 

Thank you for the quick response and for the quick fix- version 5.0.2 really resolved the problems I mentioned above. As the latest version was not updated automatically I uninstalled it and reinstalled it and it works fine now.

Can you tell me what were the default values of popup size (popup height & popup width) of version 4.9.9? I am asking because the default popup size is significantly bigger in version 5.0.2.

Tonight I will test the extension with Firefox-based browsers. If it happens that I encounter any problems I will notify you.

Thank you very much for releasing Tab Manager Plus for Firefox!

**Comment by mastef** [comment]
@smaragdus Width : 530, Height : 450
This hasn't changed. Maybe when you uninstalled it it cleared your preferences

**Comment by smaragdus** [comment]
@mastef 

> Width : 530, Height : 450

Thanks. I installed Tab Manager Plus to Firefox and so far everything seems fine.

However I noticed that diminishing the popup width & height may make some buttons invisible:

**Tab Manager Plus** version **5.0.2** - _default popup size_ (_530 x 450_) - Firefox:

![Tab Manager Plus 5 0 2 - 2019-08-06 - Firefox - 002](https://user-images.githubusercontent.com/9025757/62552393-3cf74300-b876-11e9-972d-c77b38cb854d.png)

**Tab Manager Plus** version **5.0.2** - _diminished popup size_  - Chromium:

![Tab Manager Plus 5 0 2 - 2019-08-06 - Chromium - 003](https://user-images.githubusercontent.com/9025757/62552478-644e1000-b876-11e9-8af0-5f6cc8d6e1aa.png)

Perhaps the search box should be diminished too so that all the buttons be visible? Or perhaps you should limit changing the popup size so that all the buttons should be visible? As a workaround I restored the default values to make all the buttons visible again. What about adding an extra option to restore the default size of the popup?

I am glad that Tab Manager Plus is available for Firefox too. Thank you for developing and maintaining this extremely useful extension- it is one of my favourite ones.

**Comment by mastef** [comment]
Thanks, I'll take a look

**Comment by smaragdus** [comment]
@mastef 

I posted the first review at Firefox add-ons ([here](https://addons.mozilla.org/en-US/firefox/addon/tab-manager-plus-for-firefox/reviews/)).

There is something I did not like about the latest version ([5.0.2](https://github.com/stefanXO/Tab-Manager-Plus/releases/tag/5.0.2))- before I could set the colour for the popup and it was permanent- now I have to select it time and again. Or am I missing something? I prefer to set a colour permanently for all windows- I would appreciate if there is an option for that.

**Comment by mastef** [comment]
@smaragdus What do you mean with that it's not permanent? When you re-open the tab manager popup the color is gone? Is that only in Firefox?

**Comment by smaragdus** [comment]
@mastef 

I mean that when I open a second window it has white colour- the popup retains the selected colour only the first open window. I am expecting when I set a colour the selected colour to be persistent for all windows. Perhaps I am misunderstanding the concept and the chosen colour is for a single browser window?

I am referring to the Chromium extension- Firefox is just a secondary browser and I don't use it so often. However I checked Firefox and the extension has the same behaviour- every single new window has white colour.

**Comment by mastef** [comment]
@smaragdus yes that's the wrong concept. You are giving specific colors to specific windows. So if you have 20 windows open, you can mark your email window in red, and your social window in blue. 
![image](https://user-images.githubusercontent.com/762297/62762856-4ff75680-bab4-11e9-81a9-24f8c8a0b12c.png)

**Comment by smaragdus** [comment]
@mastef 

Thanks for the clarification. I was using the colours as favourite backgrounds for the popup, not as a way to single out  separate windows. Anyway I am using mostly single window with multiple tabs (sometimes- hundreds) rather than multiple windows.

**Comment by mastef** [comment]
![image](https://user-images.githubusercontent.com/762297/62777898-0ec56d80-bad9-11e9-9ded-985adf4738da.png)

@smaragdus The horror. That's why I added the feature to limit every window to 10-15 tabs automatically 😄

**Comment by smaragdus** [comment]
@mastef 

It is really a horror and this horror was perhaps my primary reason to look for an extension like Tab **Manager Plus**, even though my Chromium-based browser has a special button:

![Cent Browser 3 9 2 45 - 2019-08-09 - button - 003](https://user-images.githubusercontent.com/9025757/62779219-3f96aa00-babb-11e9-8113-a3c8d6676cd0.png)

 I am not sure whether this button is available in Chrome as I have not used it for many years.

The limitation of tabs per window may be useful for others but for me it is not that convenient because when I do some research I may need to open a huge number of tabs.

To save unfinished work I use a session manager- I tried about a dozen and I chose **Tabsets** ([Chrome Web Store](https://chrome.google.com/webstore/detail/tabsets/ifohmndbcefggppiblfofpbkmdfmeing), [Website](https://bernhard-hofmann.github.io/Tabsets/), [GitHub](https://github.com/bernhard-hofmann/Tabsets)):

![Tabsets 1 20 - 2019-08-09 - 002](https://user-images.githubusercontent.com/9025757/62779579-4376fc00-babc-11e9-8c82-ac2deb87105c.png)


It is open source but it has an anti-feature- tracking. Perhaps you know a better session manager?

**Comment by mastef** [comment]
Hi @smaragdus, the hidden buttons in the popup should be fixed now in 5.0.3

Here's the full changelog :

```
5.0.3
=====

- Added context menu option to open Tab Manager in its' own tab
- Small fixes for icon borders not rendering properly
- Fixes for the search bar when the popup is too narrow
- Added popup title and favicon
- Disable popup auto-closing until we have a better solution
- Fixed Firefox error where a new window would not be created
```

**Comment by smaragdus** [comment]
@mastef 

Excellent work, thanks. I see that the Firefox users increase although very slowly.
